### PR TITLE
GROOVY-7014: A Java compile time error results in a runtime exception…

### PR DESCRIPTION
**Pull request by paulk-asert**
… in Groovy

### CODE REVIEW

**Comment by paulk-asert** [comment]
Dealing with list spreading is tricky. This PR makes an improvement but is still not a complete solution, so the API has a boolean to try the partial fuzzy matching. Rather than exposing another API call within ClassNode which leaks out the fuzzy boolean, I'll likely move that logic into a util class.

File: src/main/org/codehaus/groovy/control/StaticImportVisitor.java [hunk 1]
```diff
@@ -317,6 +337,10 @@ protected Expression transformMethodCallExpression(MethodCallExpression mce) {
         return result;
     }
 
+    private boolean isInnerClass(ClassNode currentClass) {
```
Comment (jwagenleitner) [inline src/main/org/codehaus/groovy/control/StaticImportVisitor.java]: There was an identical method added to `StaticVerifier`, how about replacing both with a static method in `ClassNodeUtils`?

File: src/main/org/codehaus/groovy/ast/tools/ClassNodeUtils.java [hunk 2]
```diff
@@ -65,4 +73,122 @@ public static void addDeclaredMethodMapsFromSuperInterfaces(ClassNode cn, Map<St
...
+     * Returns the property name, e.g. age, given an accessor name, e.g. getAge.
+     * Returns the original if a valid prefix cannot be removed.
+     *
+     * @param accessorName the accessor name of interest, e.g. getAge
+     * @return the property name, e.g. age, or original if not a valid property accessor name
+     */
+    public static String getPropNameForAccessor(String accessorName) {
+        int prefixLength = accessorName.startsWith("is") ? 2 : 3;
+        if (accessorName.length() < prefixLength + 1) return accessorName;
+        if (!isValidAccessorName(accessorName)) return accessorName;
```
Comment (jwagenleitner) [inline src/main/org/codehaus/groovy/ast/tools/ClassNodeUtils.java]: Would it make sense to move the length check to the `isValidAccessorName` method below?  That would ensure `isValidAccessorName` would not return true for methods named `get()`/`is()`/`set()`.

**Comment by jwagenleitner** [comment]
+1 (with just a few minor comments)

File: src/main/org/codehaus/groovy/ast/tools/ClassNodeUtils.java (hunk 2 above)
Comment (paulk-asert) [inline src/main/org/codehaus/groovy/ast/tools/ClassNodeUtils.java]: I refactored to support this. It means we calculate the prefix length an extra time in one case but the code reads better which I think is more important.

File: src/main/org/codehaus/groovy/control/StaticImportVisitor.java (hunk 1 above)
Comment (paulk-asert) [inline src/main/org/codehaus/groovy/control/StaticImportVisitor.java]: Done

**Pull request merged**